UI/script.js

```diff
@@ -185,44 +185,176 @@ const photoPosts1 = [
 
 class Controller {
     constructor() {
-        this._photoList = new PhotoList(photoPosts1/*posts from localStorage*/);
-        this._view = new View(document.getElementsByClassName('main__publications')[0], 'Guest'/*username from localStorage*/);
+        const posts = JSON.parse(localStorage.getItem('posts'));
+        if (posts && posts.length !== 0) {
+            posts.forEach((elem) => {elem.createdAt = new Date(elem.createdAt)});
+            this._photoList = new PhotoList(posts);
+        }
+        else {
+            localStorage.setItem('posts', JSON.stringify(photoPosts));
+            this._photoList = new PhotoList(photoPosts);
+        }
+        const username = localStorage.getItem('username');
+        if (username) {
+            this._view = new View(document.getElementsByClassName('main__publications')[0], username);
+        }
+        else {
+            this._view = new View(document.getElementsByClassName('main__publications')[0], 'Guest');
```

Comment:
this._view = new View(document.getElementsByClassName('main__publications')[0], username || 'Guest');